language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/runner/BallerinaRunnerService.java

```diff
@@ -0,0 +1,185 @@
+import org.ballerinalang.langserver.common.utils.PathUtil;
+import org.ballerinalang.langserver.commons.DocumentServiceContext;
+import org.ballerinalang.langserver.commons.LSOperation;
+import org.ballerinalang.langserver.commons.LanguageServerContext;
+import org.ballerinalang.langserver.commons.service.spi.ExtendedLanguageServerService;
+import org.ballerinalang.langserver.commons.workspace.WorkspaceManager;
+import org.ballerinalang.langserver.contexts.ContextBuilder;
+import org.ballerinalang.langserver.diagnostic.DiagnosticsHelper;
+import org.ballerinalang.langserver.extensions.ballerina.packages.PackageContext;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+import org.eclipse.lsp4j.jsonrpc.services.JsonSegment;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Implementation of Ballerina runner extension for Language Server.
+ *
+ * @since 2201.11.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.service.spi.ExtendedLanguageServerService")
+@JsonSegment("ballerinaRunner")
+public class BallerinaRunnerService implements ExtendedLanguageServerService {
+
+    private WorkspaceManager workspaceManager;
+    private LanguageServerContext serverContext;
+    private LSClientLogger clientLogger;
+
+    @Override
+    public void init(LanguageServer langServer, WorkspaceManager workspaceManager,
+                     LanguageServerContext serverContext) {
+        this.workspaceManager = workspaceManager;
+        this.serverContext = serverContext;
+        this.clientLogger = LSClientLogger.getInstance(serverContext);
+    }
+
+    /**
+     * Get all the diagnostics of the project.
+     *
+     * @param request {@link ProjectDiagnosticsRequest}
+     * @return {@link ProjectDiagnosticsResponse}
+     */
+    @JsonRequest
+    public CompletableFuture<ProjectDiagnosticsResponse> diagnostics(ProjectDiagnosticsRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                ProjectDiagnosticsResponse projectDiagnosticsResponse = new ProjectDiagnosticsResponse();
+                Optional<Path> filePath = PathUtil.getPathFromURI(request.getProjectRootIdentifier().getUri());
+                if (filePath.isEmpty()) {
+                    return projectDiagnosticsResponse;
+                }
+                Optional<Project> projectOptional = this.workspaceManager.project(filePath.get());
+                Project project;
+                if (projectOptional.isEmpty()) {
+                    project = this.workspaceManager.loadProject(filePath.get());
+                } else {
+                    project = projectOptional.get();
+                }
```

Comment:
The `loadProject` also returns the project if it exists in the cache. So, calling `loadProject` would be sufficient. 

```suggestion
                project = this.workspaceManager.loadProject(filePath.get());
```